### DISCUSSION TITLE
In the following query it does not return the videos

### DISCUSSION THREAD

**Discussion by xoanma** [microsoft-search/pnp-modern-search#3837] [General]
In the query template field I have the following: path:"url/*{searchTerms}*"
This should filter within the folder what I enter in the search field

**Comment by kasperbolarsen** [comment]
Please test the query in either SP Editor or SharePoint Search Query Tool first :-) 
or use the network tab to inspect the resulting query
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/8005e902-3ba6-43dd-8357-ae85f1aaef2c)

**Comment by wobba** [comment]
Using `{searchTerms}` as part of a path query will typically not work. If you paraphrase what you want to accomplish and describe your scenario we can more easily help on what KQL you need.

**Reply by xoanma** [reply]
I want to search in specific folders and be able to filter all their contents (path:"url/{searchTerms}" OR path:"url/{searchTerms}" ...). When using the path:"url" filter and searching by default with * it doesn't return all the results in the folder, as in my case it omits the .mp4 files. This really surprises me, because when searching the entire site, it does return those videos.

**Reply by kasperbolarsen** [reply]
does path:{Site} FileType="mp4" return anything ?

**Reply by xoanma** [reply]
From the main site path, I get all the MP4 results, but if I go to where any video is located and use the containing folder as the path, it doesn't return.

**Reply by kasperbolarsen** [reply]
wait a second, are you searching using this one?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ee15a29c-eb73-4a4c-88fe-28be9107fede)

**Reply by xoanma** [reply]
I am using PnP Modern Search (v4). Search box paired with Search Results.

**Comment by kasperbolarsen** [comment]
I finally think I understand the issue :-)

When looking at a specific mp4 file in the SP Editor I can see that the 
Path is https://contoso.sharepoint.com/sites/PnPModernSearch/Shared Documents/Forms/DispForm.aspx?ID=263
eventhough the file is in a folder.
I would look into using the property DocumentLink as it is this for the same file:
https://contoso.sharepoint.com/sites/PnPModernSearch/Shared Documents/Dummy docs/Folder/CustomSharePointSearch.mp4

**Reply by xoanma** [reply]
Hi Kasper,

To be honest, I am quite lost. For testing purposes, I created a folder within the site, and there it returns all results, both multimedia and documents. The case is that folders have been created for each of the departments, and here comes the weird part. When I add the full path of the folder https://xxxx.sharepoint.com/sites/xxxxx/33_SAP/33_SITE_FORMACION, it doesn't return multimedia files. However, if I use https://xxx.sharepoint.com/sites/xxx/33_SAP as the path, it does return everything. The URL I have for the training folder is the following: https://xxx.sharepoint.com/sites/xxx/33_SAP/Forms/AllItems.aspx?id=%2Fsites%2Fxxxxxx%2F33_SAP%2F33_SITE_FORMACION. I have also tried using this one with the same result. I tried making the following query: https://xxxxx.sharepoint.com/_api/search/query?querytext=%27(Path%3ahttps%3a%2f%2fxxxx.sharepoint.com%2fsites%2fxxxxxx%2f33_SAP%2f33_SITE_FORMACION)%27&enableinterleaving=false&clienttype=%27ContentSearchRegular%27, and the XML it returns brings all the results! There are folders where all files are shown, and others where they are not. I rule out that it is a permissions issue because I would never see these files, but depending on the path, they appear. The query seems to be correct as the XML shows it. Any ideas?

**Reply by xoanma** [reply]
Forget the last part, the query doesn't return it either.

**Reply by kasperbolarsen** [reply]
I would recommend using one of the tools listed here to inspect the results of the queries, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

One thing I noticed when testing this scenario is that the Path managed Property didn't contains the folder part of the URL, hence why I suggested that you might use DocumentLink  instead

**Reply by xoanma** [reply]
I have downloaded it, I'm sure it will be very useful, thank you Kasper. On another note, just as you said, I can't use the path. Looking at the properties shared by the files, I think I will use ParentLink. I tried it and it works, thanks for the help!

**Reply by kasperbolarsen** [reply]
Happy it was resolved and you are able to continue :-)
